src/Analyzers/MSTest.Analyzers/DoNotUseShadowingAnalyzer.cs

```diff
@@ -112,7 +112,8 @@ private static bool IsMemberShadowing(ISymbol member, ISymbol baseMember)
         }
 
         // Compare methods
-        if (member is IMethodSymbol methodSymbol && baseMember is IMethodSymbol baseMethodSymbol)
+        if (member is IMethodSymbol methodSymbol && baseMember is IMethodSymbol baseMethodSymbol && methodSymbol.IsGenericMethod == baseMethodSymbol.IsGenericMethod
+            && !(methodSymbol.DeclaredAccessibility == Accessibility.Private && baseMethodSymbol.DeclaredAccessibility == Accessibility.Private))
```

Comment:
It also can't be shadowing if only one of the methods is private, or can it?

Reply:
You are right, I find the syntax confusing but @engyebrahim added the not before the parenthesis so the and clause becomes a or :)

Reply:
- private and private: `!(true && true) = false` (ok)
- public and public: `!(false && false) = true` (ok)
- private and public: `!(true && false) = true` (not ok, is not shadowing)

And I think the visibility check also needs to happen in the "Compare properties" section below.

Reply:
Oh yeah good point, I overlooked last case. @engyebrahim can you please add more tests and probably fix the remaining broken part?

Reply:
public&public raises
public&private raises -> fixing that case
<img width="656" alt="image" src="https://github.com/user-attachments/assets/5ba2b3ed-c23c-43ca-8968-4d132cadb563">

private&private doesn't raise
private&public doesn't raise